core/tee/tee_svc_cryp.c

```diff
@@ -1844,6 +1844,15 @@ static TEE_Result tee_svc_cryp_check_key_type(const struct tee_obj *o,
 	case TEE_MAIN_ALGO_DH:
 		req_key_type = TEE_TYPE_DH_KEYPAIR;
 		break;
+	case TEE_MAIN_ALGO_ECDSA:
+		if (mode == TEE_MODE_VERIFY)
+			req_key_type = TEE_TYPE_ECDSA_PUBLIC_KEY;
+		else
+			req_key_type = TEE_TYPE_ECDSA_KEYPAIR;
+		break;
+	case TEE_MAIN_ALGO_ECDH:
+		req_key_type = TEE_TYPE_ECDH_KEYPAIR;
+		break;
```

Comment:
I do not think so. We only check here the key type, not all the mode.

However, I think you pointed something not supported right now: VERIFY / ENCRYPT must also work on keypair. In that case, only the public key of the keypair is taking into account. I'll submit this in another patch.